src/chainstate/stacks/db/mod.rs

```diff
@@ -1001,6 +1030,154 @@ impl StacksChainState {
                 });
             }
 
+            let bns_contract_id = boot_code_id("bns");
+            if let Some(get_namespaces) = boot_data.get_bulk_initial_namespaces.take() {
+                info!("Initializing chain with namespaces");
+                clarity_tx.connection().as_transaction(|clarity| {
+                    clarity
+                        .with_clarity_db(|db| {
+                            let initial_namespaces = get_namespaces();
+                            for entry in initial_namespaces {
+                                let namespace = {
```

Comment:
Just for safety, you may want to validate the characters in `namespace`.

Reply:
Checks added in https://github.com/blockstack/stacks-blockchain/pull/2204/commits/52579d5dd4d88987a9dbc53543c8bb85ae971cbc